### PR TITLE
Initialize cruft

### PR DESCRIPTION
**Pull request by jdrake**
Initialize cruft and template out the repo according to the cookiecutter-python-lib repo. This involves the `cruft link` command and then manually going through the diff to fix changes.

### CODE REVIEW

File: pyproject.toml
Comment (ns-bdesimone) [inline pyproject.toml:31]: Why are we pinning versions? Is this for a specific reason? Thought we were trying to move away from this?

**Review by ssantana-ns — APPROVED** [review]
Not for this PR but should we move this to python 3.9?

**Review by ns-bdesimone — APPROVED** [review]
LGTM - had one question does not block merge

**Comment by ns-bdesimone** [comment]
> Not for this PR but should we move this to python 3.9?

I am all for it... also bump the deps to latest everything possible and use "^x.x.x" so things can move on their own as long as in the same major version.

File: pyproject.toml
Comment (jdrake) [reply pyproject.toml:31]: This is restricting to a major version (which is what we want)

**Comment by jdrake** [comment]
> Not for this PR but should we move this to python 3.9?

I think you want the supported versions be >=3.6.5 but you'll have to add extra CCI stuff to test multiple python versions

> bump the deps to latest everything possible and use "^x.x.x"

This is up to @ssantana-ns -- eventually we want this but for now i bet we want to stick with the versions we have so that it's easier to use this library in the talos repo

**Pull request merged**
